shark_turbine/kernel/wave/expansion.py

```diff
@@ -178,17 +179,22 @@ def compute_index(node: fx.Node) -> bool:
 def expand_graph(
     trace: CapturedTrace,
     constraints_or_scaling: Sequence[Constraint] | dict[IndexSymbol, int],
+    tile_sizes: dict[IndexSymbol, int] = None,
 ):
     """
     Create a graph that represents the expanded version of the wave function.
     The expansion is done in the dimensions specified by the constraints.
     """
+    tile_size_mapping = {}
     if isinstance(constraints_or_scaling, dict):
         dim_scaling = constraints_or_scaling
+        dim_tile_size = tile_sizes
     else:
         mma_index = get_mma_dimensional_mapping(trace)
         set_node_indices(trace, constraints_or_scaling, mma_index)
```

Comment:
`set_node_indices` is not part of this PR, but I'll add the thought anyway:
It does an independent walk of all nodes and performs an action on them. This is required before the expansion but I am not sure it is really a part of it. Feels like a preprocessing step.

I think it will serve visibility in the future if we factor out these individual bits and call them from `wave.py`. Otherwise we will end up with huge files (this one is already almost 500 loc) that perform more than what their name indicates.

Reply:
Yes that's right. I was trying to avoid recomputing the mma_dimensional_mapping, but that is also a preprocessing step. Will move them both out of here into a separate file.

Reply:
I have modified this so that no the index calculation is part of the expansion. I think that this makes the index calculation easier to understand (with absolute indices being calculated instead of relative indices). With this, I don't think these need to be moved to a separate file.